pkg/cli/cmd/uninstall/kubernetes/kubernetes.go

```diff
@@ -113,3 +188,180 @@
 	r.Output.LogInfo("Radius was uninstalled successfully. Any existing data will be retained for future installations. Local configuration is also retained. Use the `rad workspace` command if updates are needed to your configuration.")
 	return nil
 }
+
+func (r *Runner) buildCleanupPlan(ctx context.Context, state helm.InstallState) (cleanupPlan, error) {
+	plan := cleanupPlan{}
+	plan.HelmReleases = append(plan.HelmReleases, "radius")
+	if state.ContourInstalled {
+		plan.HelmReleases = append(plan.HelmReleases, "contour")
+	}
+
+	if !r.Purge {
+		return plan, nil
+	}
+
+	plan.CRDs = append(plan.CRDs, radiusCRDs...)
+	plan.APIServices = append(plan.APIServices, ucpAPIServiceName)
+	plan.Namespaces = append(plan.Namespaces, helm.RadiusSystemNamespace, daprSystemNamespace)
+
+	environments, err := r.fetchEnvironmentCleanupInfos(ctx)
+	if err != nil {
+		plan.EnvDiscoveryFailed = true
+		return plan, fmt.Errorf("unable to enumerate Radius environments via Radius APIs: %w", err)
+	}
+	plan.Environments = environments
+
+	return plan, nil
+}
+
+func (r *Runner) describeCleanupPlan(plan cleanupPlan) {
+	r.Output.LogInfo("About to uninstall Radius. This will remove:")
+	if len(plan.HelmReleases) > 0 {
+		r.Output.LogInfo("- Helm releases: %s", strings.Join(plan.HelmReleases, ", "))
+	}
+
+	if r.Purge {
+		if len(plan.Environments) > 0 {
+			r.Output.LogInfo("- Radius environments:")
+			for _, env := range plan.Environments {
+				r.Output.LogInfo("  • %s (namespace %s)", env.ID, env.Namespace)
+			}
+		} else if plan.EnvDiscoveryFailed {
+			r.Output.LogInfo("- Radius environments: unable to determine (Radius management APIs unreachable)")
+		} else {
+			r.Output.LogInfo("- Radius environments: none")
+		}
+
+		if len(plan.Namespaces) > 0 {
+			r.Output.LogInfo("- Kubernetes namespaces: %s", strings.Join(plan.Namespaces, ", "))
+		}
+		if len(plan.APIServices) > 0 {
+			r.Output.LogInfo("- Kubernetes API services: %s", strings.Join(plan.APIServices, ", "))
+		}
+		if len(plan.CRDs) > 0 {
+			r.Output.LogInfo("- Kubernetes custom resource definitions: %s", strings.Join(plan.CRDs, ", "))
+		}
+	}
+}
+
+func (r *Runner) fetchEnvironmentCleanupInfos(ctx context.Context) ([]environmentCleanup, error) {
+	workspace := workspaces.MakeFallbackWorkspace()
```

Comment:
Workspace creation logic is duplicated. Extract to a helper function like `createWorkspaceForKubeContext()`.